src/main/java/picocli/CommandLine.java

```diff
@@ -2007,15 +2007,35 @@ private static int handleUnhandled(Exception ex, CommandLine cmd, int defaultExi
      * @param existingColorScheme the {@code ColorScheme} to use
      * @return converted and colored {@code String}
      */
-    private static String exceptionToColorString(Exception ex, Help.ColorScheme existingColorScheme){
+    private static String exceptionToColorString(Exception ex, Help.ColorScheme existingColorScheme) {
         Help.ColorScheme colorScheme = new Help.ColorScheme.Builder(existingColorScheme).applySystemProperties().build();
-        String newLine = System.getProperty("line.separator");
-        StringBuilder stringBuilder = new StringBuilder();
-        stringBuilder.append(colorScheme.errorText(ex.toString())).append(newLine);
-        for (StackTraceElement traceElement : ex.getStackTrace()) {
-            stringBuilder.append(colorScheme.stackTraceText("\tat " + traceElement)).append(newLine);
+        StringWriter stringWriter = new ColorSchemedStringWriter(colorScheme);
+        ex.printStackTrace(new PrintWriter(stringWriter));
+        return stringWriter.toString();
+    }
+
+    /**
+     * Extends StringWriter to use ColorScheme. Allows separating
+     * exception messages from stack traces by intercepting write method.
+     */
+    static class ColorSchemedStringWriter extends StringWriter {
+
+        Help.ColorScheme colorScheme;
+
+        public ColorSchemedStringWriter(Help.ColorScheme colorScheme) {
+            this.colorScheme = colorScheme;
+        }
+
+        public ColorSchemedStringWriter(int initialSize, Help.ColorScheme colorScheme) {
```

Comment:
Perhaps we can omit this constructor.

Reply:
The `initialSize` one? Seems not used by any code. I think it can be safely omitted.